### ISSUE TITLE
NOT deleted There's been an error. Your item might not have been deleted.

### ISSUE THREAD

**Issue by tedsecretsource** [Laravel-Backpack/PermissionManager#135]
# Bug report (maybe)

Given I am looking at the list view of a CRUD object
When I click the Delete button to delete a row
Then the system responds with an error message saying the row might not have been deleted (and indeed it wasn't)

### What I did:
Click the Delete button for an item.

### What I expected to happen:
The item would be deleted and the green confirmation would appear

### What happened:
The error "NOT deleted There's been an error. Your item might not have been deleted." is displayed and the item was not deleted.

### What I've already tried to fix it:
Everything… We're running this on two different homestead machines. On one of them (the Macintosh) it works fine, on the other (Ubuntu Linux) it doesn't. We're also testing on a CentOS machine (SiteGround) and it fails there too.

We've tried completely trashing and rebuilding the project to no avail.

Apache is showing 500 errors in its logs for DELETE methods on the CentOS machine (on this end point, it returns 404 for known files), which is suspicious, but the Laravel PHP error log shows no hint of seg faults or any such thing normally associated with a 500 error.

And to be clear, on CentOS (SiteGround), nginx sits on top of apache as a sort of reverse proxy. On both the homestead machines we're using straight nginx.

### Backpack, Laravel, PHP, DB version:
Backpack: 3.3.3
Laravel 5.5
PHP 7.1.12 and 7.1.5
MySQL 5.7.17

I'm seeing this in the error log whenever this happens:

`ERROR: Unauthorized access - you do not have the necessary permissions to see this page. {"userId":1,"email":"ted@secret-source.eu","exception":"[object] (Backpack\\CRUD\\Exception\\AccessDeniedException(code: 0): Unauthorized access - you do not have the necessary permissions to see this page. at /home/useraccount/projects/project.com/vendor/backpack/crud/src/PanelTraits/Access.php:90)`

Followed by a loooong stack trace. The first item in the trace is this:

`#0 /home/useraccount/projects/project.com/vendor/backpack/crud/src/app/Http/Controllers/CrudController.php(199): Backpack\\CRUD\\CrudPanel->hasAccessOrFail('show')`

Any tips or pointers on where else to look to diagnose this would be greatly appreciated. I don't really trust that hasAccessOrFail() is at the root of the problem, though, becase the code is exactly the same on all the machines (in theory - with composer I'm never really sure of that).

Thanks in advance.

PS: I tried the fix [here](https://stackoverflow.com/questions/42621376/laravel-5-4-backpack-cant-delete-item-returns-403-forbidden-error) and that silenced the error and stack trace in the Laravel PHP log, but it still didn't work. SiteGround assures me that the DELETE method is supported.

PPS: [This](https://github.com/Laravel-Backpack/CRUD/pull/989) looks like it might be related but I'm not sure how.

**Comment by tedsecretsource** [comment]
More information following additional testing…

This is only happening when making requests over HTTP**S**. If I disable redirection to HTTPS, deleting works. This is what I've observed so far:

With HTTPS…

The permission sent to hasAccessOrFail is 'show' and not 'delete'

With HTTP…

The permission sent to hasAccessOrFail is 'delete'

Given that some assets in the CRUD manager are being loaded over non-HTTPS, I have to suspect that this might be happening due to cross-origin something-or-other failing to forward the DELETE method. However, I *am* seeing the DELETE method in the server access logs (which is returning a 500 error) so I'm really at a loss on this. 

Also, I'm using HTTPS on all of our machines (two homestead machines for development and a CentOS machine as the server). The delete fails on the linux-based machines but not on the Macintosh based development machine.

Any help would be greatly appreciated.

**Comment by tedsecretsource** [comment]
SiteGround's web hosting environment (on dedicated servers) does not seem to support the DELETE method under HTTPS. To test, I captured the value of $_SERVER['REQUEST_METHOD'] under HTTP (DELETE) and under HTTPS (GET). [This workaround DID solve the problem](https://stackoverflow.com/questions/42621376/laravel-5-4-backpack-cant-delete-item-returns-403-forbidden-error) (I missed the last line the first time I tried it).

As today is Christmas day, consider this a present to whoever else stumbles upon this problem. And, just because, I'm submitting a support request to SiteGround to see if we can get this changed.
